dev-ruby/grpc/grpc-1.55.1.ebuild

```diff
@@ -0,0 +1,79 @@
+# Copyright 2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+USE_RUBY="ruby30 ruby31 ruby32"
+
+# RUBY_FAKEGEM_RECIPE_TEST="rspec3"
+# RUBY_FAKEGEM_EXTRADOC="README.md"
+RUBY_FAKEGEM_EXTENSIONS="ext/grpc/extconf.rb"
+RUBY_FAKEGEM_EXTENSION_LIBDIR="ext/grpc/lib/${PN}"
+RUBY_FAKEGEM_GEMSPEC="../../${PN}.gemspec"
+RUBY_S="${P}/src/ruby"
+
+inherit ruby-fakegem
+
+DESCRIPTION="Ruby implementation of gRPC"
+HOMEPAGE="https://grpc.io/"
+SRC_URI="https://github.com/grpc/grpc/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0/32.155"
+KEYWORDS="~amd64"
+
+RDEPEND="
```

Comment:
I have read the docs, but would you be able to better differentiate DEPEND and RDEPEND for me? Does DEPEND cover both build and runtime?

Reply:
BDEPEND - tools executed during the build (think e.g. `msgfmt`, `pkg-config`, `doxygen`, ...). This gets fuzzier for scripting languages like Ruby and Python and it only really matters if there's C extensions being built.
DEPEND - headers and libraries needed during the build
RDEPEND - runtime. In theory, could be absent while the package is built, but we don't do that right now.

Does that help or should I elaborate a bit more?

Reply:
My main confusion was about how DEPEND and BDEPEND relate to each other (though I notice that I wrote RDEPEND up above). So BDEPEND is more tools used at build time where DEPEND is headers and libraries for includes and linking, thus they will be required at runtime too (should the the libraries be shared rather than static and built in) and RDEPEND would be tools required at runtime, though not linked against. As long as my rewording fits, I think you have explained it well enough. Thank you.

Reply:
RDEPEND includes everything required at runtime, including libraries linked against.  Most libraries are both in DEPEND and RDEPEND because they are required both at build time and at runtime.